site/src/pages/UserSettingsPage/SSHKeysPage/SSHKeysPage.tsx

```diff
@@ -46,17 +40,15 @@ export const SSHKeysPage: FC<PropsWithChildren<unknown>> = () => {
       <ConfirmDialog
         type="delete"
         hideCancel={false}
-        open={authState.matches("signedIn.ssh.loaded.confirmSSHKeyRegenerate")}
-        confirmLoading={authState.matches(
-          "signedIn.ssh.loaded.regeneratingSSHKey",
-        )}
+        open={sshState.matches("confirmSSHKeyRegenerate")}
```

Comment:
Sometimes we're making variables for sshState keys and sometimes we aren't - can we choose one and be consistent?

Reply:
The matches function only return true or false. Unfortunately, xstate v4 does not have a way to bind a context type with a specific state so, sometimes, we have to check the state with match and pass the value in the context. Buuut, after these minor refactoring, I think we can remove some xstate machines and use just use fetch directly.